packages/neon-api/src/notifications/subscription.ts

```diff
@@ -0,0 +1,32 @@
+import { logging } from "@cityofzion/neon-core";
+const log = logging.default("api");
+
+export class Subscription {
+  private contract: string | null;
+  private unsubscribeFunction: () => void;
+  private alive: boolean;
+
+  public get name(): string {
+    return `Subscription[${this.contract}]`;
+  }
+
+  public constructor(contract: string | null, unsubscribeFunction: () => void) {
+    this.contract = contract;
+    this.unsubscribeFunction = unsubscribeFunction;
+    this.alive = true;
+    log.info(`Created Subscription: ${this.contract}`);
+  }
+
+  /**
+   * Unsubscribe a specific function from a contract
+   */
+  public unsubscribe() {
+    if (!this.alive) {

```

Comment:
this inner variable feels redundant when we are not throwing. The savings is pretty small for the cost of having a possible wrong state.

Reply:
What alternative do you think would work best? Throwing exceptions?

Reply:
I meant that you could just remove the `alive` check and just call `unsubscribeFunction` since it doesn't fail.

Reply:
I see what you are saying, but that could result in unexpected behaviour in the following case:
```
let s1 = notifications.subscribe("0x...", console.log)
let s2 = notifications.subscribe("0x...", console.log)
s1.unsubscribe()
s1.unsubscribe() //s2 gets unsubscribed
```

With that said, this is a sort of an extreme edge case so we may want to ignore it.

Reply:
ah thats interesting then the issue here is that we are storing the reference to the callback as an unique identifier. This solves the problem by ensuring each subscription can only at most remove one callback. A little roundabout but i guess it works.

Reply:
I've changed how that works to make reduce the edge cases and the complexity, see https://github.com/CityOfZion/neon-js/pull/514#issuecomment-626260820